### ISSUE TITLE
fireImmediately at AutodisposeStateNotifierProvider will crash if watched directly

### ISSUE THREAD

**Issue by readyiman** [rrousselGit/riverpod#3658]
I'm using `AutodisposeStateNotifierProvider` to listen on position changes. I use `StateNotifier` because (for some reason) I don't want it to rebuild if the listened provider make changes, and I enable `fireImmediately` if the listened provider may had changes. Then I watch this provider from other provider. This will crash with errors __debugCurrentlyBuildingElement == this': Providers are not allowed to modify other providers during their initialization._ . But this errors will gone if I watched this notifier first.

```dart
class PositionPs {
static final positionAP =
      StateNotifierProvider.autoDispose<_PositionN, LatLng?>((ref) {
    final noti = _PositionN();

    ref.listen(
      aPositionUpdateProviderP,
      (previous, next) {
        if (next != null) noti.change(next);
      },
      fireImmediately: true,
    );

    return noti;
  });

  // Errors will gone with this watching above
  // static final positionBP =
  //     Provider.autoDispose<LatLng?>((ref) => ref.watch(positionAP));
}

class _PositionN extends StateNotifier<LatLng?> {
  _PositionN() : super(null);

  void change(LatLng c) => state = c;
}

class MapPs {
static final _followPositionP = Provider.autoDispose<LatLng?>((ref) {
    final pos = ref.watch(PositionPs.positionAP); // Will crash, but not with positionBP
    final follow = ref.watch(isFollowingUserP);
    return follow
        ? pos
        : null;
  });
}
```

 I suspect watching it first is the correct way to do it. But why?

Note: This behaviour won't happen without `fireImmediately` enabled.

**Comment by rrousselGit** [comment]
Looks like an oversight in the assert. Will fix in 3.0

**Comment by readyiman** [comment]
I can't reproduce this after upgrading to Flutter 3.22 (was 3.19). I'm closing this.
